Add docs for 5177 feature migrate media to html utility methods to v2 sdk (#5189)
recognai/rubrix: argilla/docs/how_to_guides/use_markdown_to_format_rich_content.md

# Use Markdown to format rich content

This guide provides an overview of how to use Markdown and HTML in `TextFields` to format chat conversations and allow for basic multi-modal support for images, audio, video and PDFs.

> The `TextField` and `TextQuestion` provide the option to enable Markdown and therefore HTML. Given the flexibility of HTML, we can get great control over the presentation of data to our annotators. We provide some out-of-the-box methods for multi-modality and chat templates in the examples below.

Reviewer: ```suggestion
The `TextField` and `TextQuestion` provide the option to enable Markdown and, therefore, HTML. Given the flexibility of HTML, we can have great control over how data is presented to our annotators. We provide some out-of-the-box methods for multi-modality and chat templates below.
```

!!! note

>     We do assume that a `TextField` or `TextQuestion` has been configured with `use_markdown=True`. Take a look at [`Datasets`](dataset.md) to learn more about this configuration.

Reviewer: ```suggestion
    We assume that a `TextField` or `TextQuestion` has been configured with `use_markdown=True`. To learn more about the dataset's settings, take a look at [this guide](dataset.md).
```

!!! tip

>     You can get pretty creative with HTML. Think for example about visualizing graphs and tables. Some interesting Python package examples that allow for this are the `pandas.DataFrame.to_html` and the `plotly.io.to_html` methods.

Reviewer: ```suggestion
    You can get pretty creative with HTML. For example, think about visualizing graphs and tables. You can use some interesting Python packages methods like `pandas.DataFrame.to_html` and `plotly.io.to_html`.
```

## Multi-modal support: images, audio, video, PDFs and more

> Argilla has basic image support in different ways that both have their pros and cons but offer the same UI experience because they both rely on HTML.

Reviewer: ```suggestion
Argilla has basic multi-modal support in different ways, each with pros and cons, but they both offer the same UI experience because they both rely on HTML.
```

=== "Image"

    ```python
    from argilla.markdown import image_to_html

    html = image_to_html(
        "local_image_file.png",
        width="300px",
        height="300px"
    )

    rg.Record(
        fields={"markdown_enabled_field": html}
    )
    ```

=== "Audio"

    ```python
    from argilla.markdown import audio_to_html

    html = audio_to_html(
        "local_audio_file.mp3",
        width="300px",
        height="300px",
        autoplay=True,
        loop=True
    )

    rg.Record(
        fields={"markdown_enabled_field": html}
    )
    ```

=== "Video"

    ```python
    from argilla.markdown import video_to_thml

    html = video_to_html(
        "my_video.mp4",
        width="300px",
        height="300px",
        autoplay=True,
        loop=True
    )

    rg.Record(
        fields={"markdown_enabled_field": html}
    )
    ```

=== "PDF"

    ```python
    from argilla.markdown import pdf_to_html

    html = pdf_to_html(
        "local_pdf.pdf",
        width="300px",
        height="300px"
    )

    rg.Record(
        fields={"markdown_enabled_field": html}
    )
    ```

### Hosted content

> Instead of uploading local files through DataURLs we can also visualize URLs that link directly to media files such as images, audio, video, and PDFs hosted on a public or private server. In this case, you can use basic HTML to visualize content that is available on something like Google Photos or decide to configure a private media server.

Reviewer: ```suggestion
Instead of uploading local files through DataURLs, we can also visualize URLs directly linking to media files such as images, audio, video, and PDFs hosted on a public or private server. In this case, you can use basic HTML to visualize content available on platforms like Google Drive or decide to configure a private media server.
```

    ```python
    html = """
    <iframe
        src="https://example.com/public-pdf-file.pdf"
        width="600"
        height="500">
    </iframe>
    """"

    rg.Record(
        fields={"markdown_enabled_field": html}
    )
    ```

##  Chat and conversation support

When working with chat data from multi-turn interaction with a Large Language Model, it might be nice to be able to visualize the conversation in a similar way as a common chat interface. To facilitate this, we offer the `chat_to_html` function, which converts messages from OpenAI chat format to an HTML-formatted chat interface.

??? Question "OpenAI chat format"
    The OpenAI chat format is a way to structure a list of messages as input from users and returns a model-generated message as output. These messages can only contain the `roles` "user" for human messages and "assistant", "system" or "model" for model-generated messages.

```python
from argilla.markdown import chat_to_html

messages = [
    {"role": "user", "content": "Hello! How are you?"},
    {"role": "assistant", "content": "I'm good, thank you!"}
]

html = chat_to_html(messages)

rg.Record(
    fields={"markdown_enabled_field": html}
)
```

> ![media](../assets/images/how_to_guides/markdown/chat.png)

Reviewer: Maybe add something like the other guides:
> Check the [Markdown - Python Reference](https://argilla-io.github.io/argilla/latest/reference/argilla/users/) to see these methods in detail.
